src/lvl_script_value.c

```diff
@@ -165,52 +165,37 @@ TbResult script_use_power_on_creature_matching_criterion(PlayerNumber plyr_idx,
 TbResult script_use_spell_on_creature(PlayerNumber plyr_idx, ThingModel crmodel, long criteria, long fmcl_bytes)
 {
     struct Thing *thing = script_get_creature_by_criteria(plyr_idx, crmodel, criteria);
-    if (thing_is_invalid(thing)) {
-        SYNCDBG(5,"No matching player %d creature of model %d (%s) found to use spell on.",(int)plyr_idx,(int)crmodel, creature_code_name(crmodel));
+    if (thing_is_invalid(thing))
+    {
+        SYNCDBG(5, "No matching player %d creature of model %d (%s) found to use spell on.", (int)plyr_idx, (int)crmodel, creature_code_name(crmodel));
         return Lb_FAIL;
     }
-    SpellKind spkind = (fmcl_bytes >> 8) & 255;
-    const struct SpellConfig* spconf = get_spell_config(spkind);
-
-    if (spconf->caster_affected ||
-            (spkind == SplK_Freeze) || (spkind == SplK_Slow) || // These two should be also marked at configs somehow?
-            ( (spkind == SplK_Disease) && ((get_creature_model_flags(thing) & CMF_NeverSick) == 0) ) ||
-            ( (spkind == SplK_Chicken) && ((get_creature_model_flags(thing) & CMF_NeverChickens) == 0) ) )
-    {
+    SpellKind spkind = (fmcl_bytes >> 8) & 255; // What the hell is this?
+    struct SpellConfig *spconf = get_spell_config(spkind);
+    if ((spconf->caster_affected) && (!creature_is_immune_to_spell_flags(thing, spconf->spell_flags)))
+    { // Immunity is handled in 'apply_spell_effect_to_thing', but this command plays sounds, so check for it.
         if (thing_is_picked_up(thing))
         {
-            SYNCDBG(5,"Found creature to cast the spell on but it is being held.");
+            SYNCDBG(5, "Found creature to cast the spell on but it is being held.");
             return Lb_FAIL;
         }
         unsigned short sound;
-        if (spconf->caster_affected)
+        if (spconf->caster_affect_sound)
```

Comment:
Are you sure this change is correct. Are you now breaking the option to set the sound to 0?

Reply:
I think it is fine to have no sound on a self cast.